src/joyride/sci.cljs

```diff
@@ -31,6 +32,28 @@
       {:file ns-path
        :source (str (fs/readFileSync path-to-load))})))
 
+(def ^:private extension-namespace-prefix "ext://")
+
+(defn- extract-extension-name [namespace]
+  (subs namespace (count extension-namespace-prefix)))
+
+(defn- active-extension? [namespace]
+  (when (.startsWith namespace extension-namespace-prefix)
+    (let [[extension-name _module-name] (str/split (extract-extension-name namespace) #"\$")
+          extension (vscode/extensions.getExtension extension-name)]
+      (and extension
+           (.-isActive extension)))))
+
+(defn- extension-module [namespace]
+  (let [[extension-name module-name] (str/split (extract-extension-name namespace) #"\$")
+        extension (vscode/extensions.getExtension extension-name)]
+    (when extension
```

Comment:
Should joyride give an error when the extension isn't found?

Reply:
I guess now that it is very explicit that it is an extension we are reaching for, that it could make sense with an error.

However, maybe we should not croak? As a user you enable/disable/uninstall extensions a bit at will. It would be up to the user to check that an extension is active before exercising its API. How about an error message written to the console? Maybe to the Joyride output channel as well?

Reply:
I think it will croak anyway if you use `:as` or `:refer` and then do interop on that.

Reply:
What about extension loading order btw? What if you require an extension that hasn't been loaded yet, is that possible?

Reply:
Right now we will simply not require it if it is not active.

> I think it will croak anyway if you use :as or :refer and then do interop on that.

We won't be doing the `:refer`s on a non-active extension.

But when the script starts trying to use the API things will blow up if the extension is not active since it will not be required. This is what I mean with that it will be up to the user to guard against this. Unless we croak instead of just skipping the require. But that is less flexible for the user, I think.

Reply:
My intuition would be to croak if you _require_ a library and that library is missing, but we can fix that later.